### ISSUE TITLE
Build error in Arch Linux:Cannot find module app.asar/build/electron.js

### ISSUE THREAD

**Issue by zxp19821005** [kevinfrei/EMP#59]
I want port it to Arch Linux.
When I build the program,I got an error:
```
$ electron-builder
  • electron-builder  version=23.6.0 os=6.1.31-1-lts
  • loaded configuration  file=package.json ("build" field)
  • loaded parent configuration  preset=react-cra
  • writing effective config  file=dist/builder-effective-config.yaml
  • packaging       platform=linux arch=x64 electron=23.1.4 appOutDir=dist/linux-unpacked
  ⨯ Application entry file "build/electron.js" in the "${HOME}/.cache/paru/clone/EMP-0.8.7/dist/linux-unpacked/resources/app.asar" does not exist. Seems like a wrong configuration.  failedTask=build stackTrace=Error: Application entry file "build/electron.js" in the "${HOME}/.cache/paru/clone/EMP-0.8.7/dist/linux-unpacked/resources/app.asar" does not exist. Seems like a wrong configuration.
    at error (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/asar/asarFileChecker.ts:7:12)
    at Object.checkFileInArchive (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/asar/asarFileChecker.ts:31:11)
    at LinuxPackager.checkFileInPackage (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/platformPackager.ts:503:7)
    at LinuxPackager.sanityCheckPackage (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/platformPackager.ts:551:5)
    at LinuxPackager.doPack (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/platformPackager.ts:312:5)
    at LinuxPackager.pack (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/platformPackager.ts:136:5)
    at Packager.doBuild (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/packager.ts:441:9)
    at Object.executeFinally (/usr/lib/node_modules/electron-builder/node_modules/builder-util/src/promise.ts:12:14)
    at Packager._build (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/packager.ts:376:31)
    at Packager.build (/usr/lib/node_modules/electron-builder/node_modules/app-builder-lib/src/packager.ts:337:12)
    at Object.executeFinally (/usr/lib/node_modules/electron-builder/node_modules/builder-util/src/promise.ts:12:14)
```
When I execute the following command in the `package.json`:
`yarn prep-release && yarn opt-release && electron-builder --publish=always`
I got this,and there was no errors:
```
$ yarn prep-release && yarn opt-release && electron-builder --publish=always
[1]  tsc -p config/tsconfig.render.rel.json exited with code 0
[0] static/main/AudioDatabase.ts(29,56): error TS2307: Cannot find module 'shared' or its corresponding type declarations.
[0] static/main/AudioDatabase.ts(69,13): error TS2339: Property 'forEach' does not exist on type '{}'.
[0] static/main/AudioDatabase.ts(69,24): error TS7031: Binding element 'type' implicitly has an 'any' type.
[0] static/main/AudioDatabase.ts(69,30): error TS7031: Binding element 'value' implicitly has an 'any' type.
[0] static/main/Communication.ts(6,52): error TS2307: Cannot find module 'shared' or its corresponding type declarations.
[0] static/main/Transcoding.ts(16,8): error TS2307: Cannot find module 'shared' or its corresponding type declarations.
[0] static/main/menu.ts(13,42): error TS2307: Cannot find module 'shared' or its corresponding type declarations.
[0]  tsc -p config/tsconfig.static.rel.json exited with code 2
[
  {
    command: Command {
      close: [Subject],
      error: [Subject],
      stdout: [Subject],
      stderr: [Subject],
      timer: [Subject],
      killed: false,
      exited: true,
      index: 1,
      name: '',
      command: ' tsc -p config/tsconfig.render.rel.json',
      prefixColor: '',
      env: {},
      cwd: '',
      killProcess: [Function (anonymous)],
      spawn: [Function (anonymous)],
      spawnOpts: [Object],
      process: undefined,
      pid: 248340,
      stdin: [Socket]
    },
    index: 1,
    exitCode: 0,
    killed: false,
    timings: {
      startDate: 2023-06-02T09:03:37.779Z,
      endDate: 2023-06-02T09:03:38.835Z,
      durationSeconds: 1.056815719
    }
  },
  {
    command: Command {
      close: [Subject],
      error: [Subject],
      stdout: [Subject],
      stderr: [Subject],
      timer: [Subject],
      killed: false,
      exited: true,
      index: 0,
      name: '',
      command: ' tsc -p config/tsconfig.static.rel.json',
      prefixColor: '',
      env: {},
      cwd: '',
      killProcess: [Function (anonymous)],
      spawn: [Function (anonymous)],
      spawnOpts: [Object],
      process: undefined,
      pid: 248339,
      stdin: [Socket]
    },
    index: 0,
    exitCode: 2,
    killed: false,
    timings: {
      startDate: 2023-06-02T09:03:37.776Z,
      endDate: 2023-06-02T09:03:39.261Z,
      durationSeconds: 1.48595283
    }
  }
]
```
When I execute the `emp` in `dist/linux-unpacked`,I got this:
```
$ ./dist/linux-unpacked/emp 
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module '${HOME}/.cache/paru/clone/EMP-0.8.7/dist/linux-unpacked/resources/app.asar/build/electron.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1002:15)
    at n._resolveFilename (node:electron/js2c/browser_init:2:109797)
    at node:electron/js2c/browser_init:2:115663
    at node:electron/js2c/browser_init:2:115906
    at node:electron/js2c/browser_init:2:115910
    at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/loaders:331:7)
    at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/loaders:269:10)
    at loadBuiltinModule (node:internal/modules/cjs/helpers:50:9)
    at Module._load (node:internal/modules/cjs/loader:862:15)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
```
Will you please fixing it ? Thank you!

**Comment by kevinfrei** [comment]
You need to to do a `yarn build-shared` first. There's a sub-module that lives in the 'shared' directory that the UI and Main processes both depend on. I'm in the middle of migrating to vite for building. Once I'm all done with that, I might go try to make the shared thing work better and be more integrated into the normal build, so you don't hit this issue :/

**Comment by zxp19821005** [comment]
```
$ yarn run build-shared 
Usage Error: Couldn't find the node_modules state file - running an install might help (findPackageLocation)

$ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] <scriptName> ...
```
I got this error when run `yarn build-shared`.

**Comment by kevinfrei** [comment]
Your troubles were enough to push me to migrate away from my original cobbled-together build to vite, yarn workspaces, and lerna. So now, if you `yarn install` then `yarn build` everything _should_ work. The github CI build appears to be working on all 3 platforms (just not releasing: I'm digging in to that). I assume that Arch Linux isn't dramatically different from any other random distro I've used (I've built it for both x86 and arm64 linux before, on some weirdo ARM & x86 devices/distros, so it ought to work). If you still have troubles, I'll dig out a box to install Arch Linux on and try to diagnose any more issues.
